scripts/asm-installer/tests/run_migration_suite_citadel

```diff
@@ -119,6 +119,19 @@ main() {
   verify_demo_app "${GATEWAY}" || SUCCESS=1
   set -e
 
+  if [[ "${SUCCESS}" -eq 1 ]]; then
```

Comment:
I'm curious if you could extract all this as a helper function instead of duplicating it in each test suite.

Reply:
I agree with all of these, but fixing these four wouldn't do much to solve the underlying issue that the tests are just really rough around the edges. I filed #165 as a followup to make the tests match the script style overall.